### PR TITLE
Fix isPlayerHudComponentVisible incompatible with command `showhud`

### LINKED ISSUE #547: isPlayerHudComponentVisible only works with setPlayerHudComponentVisible
**Describe the bug**
when Using isPlayerHudComponentVisible( string Component) returns always true until using setPlayerHudComponentVisible

Examples when it is wrong:
1. User uses showHud command there is no Hud but isPlayerHudComponentVisible says there is a hud visible
2. the breath is always true but when a player is not in water there is no breath in the hud
3. sing Camerafunctions like setCameraTarget oder setCameraMatrix always hide Health, Money, Weapon... but isPlayerHudComponentVisible says it is visible.

**To reproduce**
- `showhud`
- `crun isPlayerHudComponentVisible("health")` returns true when should be false

**Expected behaviour**
To return false when component not on screen.

**Screenshots**

**MTA Client:** 
 - Version: 
 - Running in Wine? 

**MTA Server:** 
 - Version 
 - OS: 

**Additional context**
Add any other context about the problem here

From https://bugs.mtasa.com/view.php?id=7356

### PR DESCRIPTION
**Pull request by Unde-R**
this should Fix #547 since the command `showhud` disables the hud and `isPlayerHudComponentVisible` only checks if the component is visible so It makes no sense. 
This should be enough but feel free to say anything.

### CODE REVIEW

**Comment by patrikjuvonen** [comment]
> This is a weird one. There needs to be a distinction between "visible" and "enabled"...

_Originally posted by @qaisjp in https://github.com/multitheftauto/mtasa-blue/issues/547#issuecomment-423599054_

If I understand correctly this pull request would make it impossible to know whether a component has been disabled by script or user. I think this is a negative side effect that must be solved before merge. Also, maybe allow getting `showhud` state using another function or make new function to know if it's toggled by the user (maybe there is a function already?)

**Comment by Unde-R** [comment]
Uhh.. maybe we can work on `CCommandFuncs::HUD`

**Review by qaisjp — CHANGES_REQUESTED** [review]
I agree we can't merge the current version of this. I've explained why at https://github.com/multitheftauto/mtasa-blue/issues/547#issuecomment-671595861.

Let's discuss the solution on #547 and we've settled on a solution you can implement it here if you want.

**Comment by Unde-R** [comment]
Here is the result:
[`https://pastebin.com/mAqf4hmU`](https://pastebin.com/mAqf4hmU)

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1954]: This, and elsewhere, should be `bCheckEnabled`.

File: Client/game_sa/CHudSA.cpp
Comment (qaisjp) [inline Client/game_sa/CHudSA.cpp:283]: Why not use `if (bIsEnabled && ...->IsDisabled())`?

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp
Comment (ghost) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:1958]: Since you're modifying this function, it's probably better to remove it. We don't put code in `CStaticFunctionDefinitions` anymore. You can call `g_pGame->GetHud()->IsComponentVisible` directly. The rest of the PR looks okay.

**Review by qaisjp — COMMENTED** [review]
Had these two comments from a draft review ages ago, I guess I forgot to submit

File: Client/game_sa/CHudSA.cpp
Comment (Unde-R) [reply Client/game_sa/CHudSA.cpp:283]: yeah sure

**Pull request closed without merging**
